src/serializers/type_serializers/float.rs

```diff
@@ -73,12 +73,18 @@ impl TypeSerializer for FloatSerializer {
         let py = value.py();
         match extra.ob_type_lookup.is_type(value, ObType::Float) {
             IsType::Exact => Ok(value.into_py(py)),
-            IsType::Subclass => match extra.mode {
-                SerMode::Json => {
-                    let rust_value = value.extract::<f64>()?;
-                    Ok(rust_value.to_object(py))
+            IsType::Subclass => match extra.check {
+                SerCheck::Strict => {
+                    extra.warnings.on_fallback_py(self.get_name(), value, extra)?;
+                    infer_to_python(value, include, exclude, extra)
                 }
-                _ => infer_to_python(value, include, exclude, extra),
+                SerCheck::Lax | SerCheck::None => match extra.mode {
+                    SerMode::Json => {
+                        let rust_value = value.extract::<f64>()?;
+                        Ok(rust_value.to_object(py))
+                    }
+                    _ => infer_to_python(value, include, exclude, extra),
+                },
```

Comment:
Do we still need the special `extra.mode` handling on the `SerCheck::Strict` branch too?

Reply:
I don't think so - shouldn't we always warn there bc it's strict?

Reply:
Right I see, inside `on_fallback_py` we are guaranteed to hit the `Err` condition. Can we instead refactor so that we return `Err` immediately on line 78 then, so the control flow is more explicit?

Reply:
Good call, I've modified the code accordingly!